modules/ppcp-axo/resources/js/AxoManager.js

```diff
@@ -162,6 +162,20 @@ class AxoManager {
                 ev.preventDefault();
             }
         });
+
+        // Listening to status update event
+        document.addEventListener('axo_status_updated', (ev) => {
+            const termsField = document.querySelector("[name='terms-field']");
+            if(termsField) {
+                const status = ev.detail;
+
+                if(status.active && status.validEmail === false && status.hasProfile === false) {
+                    termsField.parentElement.style.display = 'none';
+                } else {
+                    termsField.parentElement.style.display = 'block';
+                }
```

Comment:
Nitpicking, but perhaps we can improve the readability of the code a bit with something like:

```suggestion
                const shouldHide = status.active && !status.validEmail && !status.hasProfile;

                termsField.parentElement.style.display = shouldHide ? 'none' : 'block';
```